### ISSUE TITLE
vivify_flush_smaller is not strictly weakly ordered

### ISSUE THREAD

**Issue by deiruch** [arminbiere/cadical#20]
In its current form, `vivify_flush_smaller` violates its contract, as the [C++ spec requires](https://en.cppreference.com/w/cpp/named_req/Compare) a comparison function with ["strict weak ordering"](https://en.wikipedia.org/wiki/Weak_ordering#Strict_weak_orderings). Otherwise, some runtimes hit an assert in debug builds, for example VC++ 2019.

I believe the line
https://github.com/arminbiere/cadical/blob/45029f8fe26c828952c460ab4862ab28d6181484/src/vivify.cpp#L293

should be changed to
```cpp
         return j == eob && i != eoa;
```

**Comment by a1880** [comment]
The changed code would be:

```
    const auto eoa = a->end(), eob = b->end();
    auto j = b->begin();
    auto i = a->begin();
    for (; i != eoa && j != eob; i++, j++)
        if (*i != *j) return *i < *j;

    //  return j == eob;
    return j == eob && i != eoa;
```
Resulting effect of the change is that comparisons of equal clauses are flipped.

In terms of integer arrays:

Current situation:
```
[1, 2] < [1, 2]
[1, 2] < [1]
[1] >= [1, 2]
[1] < [1]
[1] < []
[] >= [1]

```

Situation with the change:
```
[1, 2] >= [1, 2]
[1, 2] < [1]
[1] >= [1, 2]
[1] >= [1]
[1] < []
[] >= [1]

```
In my VC++ 2019 build, not assertion is violated with or without the change.

**Comment by arminbiere** [comment]
Thanks for pointing this out.  Did not see the 'strict' part in the definition for 'stable_sort' (as you can see from the context, this code is used to fix some issues with non stable sorting, so this is then a fix of the fix ;-)

Will be fixed in the next release.

**Comment by arminbiere** [comment]
Fixed in release 1.3.0.
